### PR TITLE
feat: enabled sharded gitaly & added configuration docs

### PR DESCRIPTION
**Pull request by JoeHCQ1**
- Added single variable to enable gitaly sharding
- Added docs with multiple links to official docs to guide administrators including "proof of life" that this sharding works in dev
- Tested in dev as implied by the above docs

### CODE REVIEW

File: bundles/eksd/uds-bundle.yaml
Comment (MxNxPx) [inline bundles/eksd/uds-bundle.yaml:1]: this file can go now since we removed eksd in a previous PR

File: docs/gitlab.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Configuration Notes
+
+Collected notes to aid in configuring GitLab for your environment.
+
+## Sharded Gitaly
+
+[Gitaly](https://docs.gitlab.com/ee/administration/gitaly/), the git server inside GitLab, has a few HA configurations. The Gitaly cluster (where multiple Gitaly nodes are clustered together, each one as the sole-tenant on a server/VM) is the recommended approach but is not supported on k8s (see [epic](https://gitlab.com/groups/gitlab-org/-/epics/6127) for more details). Sharded Gitaly does work on k8s though. With Gitaly sharded, the gitaly statefulset will increase it's number of replicas with each one becoming a different storage path for GitLab.Per the weights set by the administrator in the GitLab UI, GitLab will roll-the-dice to determine which gitaly recieves each repo as it's created. It is on the administrators to rebalance nodes when random distributions lead to poor distribution of traffic. Read about [Gitaly capabilities](https://docs.gitlab.com/ee/administration/gitaly/gitaly_geo_capabilities.html#gitaly-capabilities) for more information on the trade-offs and risks between Gitaly Clusters and Gitaly Shards.
```
Comment (MxNxPx) [inline docs/gitlab.md:7]: nit:
`GitLab.Per` -> `Gitlab. Per`
`recieves` -> `receives`

```suggestion
[Gitaly](https://docs.gitlab.com/ee/administration/gitaly/), the git server inside GitLab, has a few HA configurations. The Gitaly cluster (where multiple Gitaly nodes are clustered together, each one as the sole-tenant on a server/VM) is the recommended approach but is not supported on k8s (see [epic](https://gitlab.com/groups/gitlab-org/-/epics/6127) for more details). Sharded Gitaly does work on k8s though. With Gitaly sharded, the gitaly statefulset will increase it's number of replicas with each one becoming a different storage path for GitLab. Per the weights set by the administrator in the GitLab UI, GitLab will roll-the-dice to determine which gitaly receives each repo as it's created. It is on the administrators to rebalance nodes when random distributions lead to poor distribution of traffic. Read about [Gitaly capabilities](https://docs.gitlab.com/ee/administration/gitaly/gitaly_geo_capabilities.html#gitaly-capabilities) for more information on the trade-offs and risks between Gitaly Clusters and Gitaly Shards.
```

Comment (JoeHCQ1) [reply docs/gitlab.md:7]: "i before e" except in about half the words in the english dictionary :D

**Review by jacobbmay — APPROVED** [review]
LGTM

**Pull request merged**